src/ServerConfig.cc

```diff
@@ -161,6 +179,7 @@ const std::string &ServerConfig::PluginInfo::Filename() const
 //////////////////////////////////////////////////
 void ServerConfig::PluginInfo::SetFilename(const std::string &_filename)
 {
+  this->dataPtr->plugin.SetFilename(_filename);
   this->dataPtr->filename = _filename;
```

Comment:
Is there a need to keep the separate variables? We could have trouble keeping them in sync

Reply:
Yes, the signature of `sdf::Plugin::Filename()` is not compatible with `ServerConfig::PluginInfo::Filename()`